### ISSUE TITLE
Building targets AMD, aarch Tablegen code does not compile 

### ISSUE THREAD

**Issue by mahesh-attarde** [llvm/llvm-project#87908]
I have change that spans over multiple targets, few of those targets produce following compile error for tablegen generated code.
I am using 

```
cmake -G "Unix Makefiles" -DDEBUG_LEVEL=1 -DCMAKE_BUILD_TYPE=Debug  -DLLVM_ENABLE_WERROR=ON  -DLLVM_TARGETS_TO_BUILD="AMDGPU;ARM;AVR;BPF;Hexagon;Lanai;LoongArch;Mips;MSP430;NVPTX;PowerPC;RISCV;Sparc;SystemZ;VE;WebAssembly;X86;XCore"   -S ../llvm
```
Error is
```
llvm-project/build/lib/Target/AMDGPU/AMDGPUGenSearchableTables.inc: In function ‘const llvm::AMDGPU::MIMGDimInfo* llvm::AMDGPU::getMIMGDimInfoByEncoding(uint8_t)’:
/iusers/mattarde/lorg/llvm-project/build/lib/Target/AMDGPU/AMDGPUGenSearchableTables.inc:2351: error: comparison is always false due to limited range of data type [-Werror=type-limits]
 2351 |   if ((Encoding < 0x0) ||
      | 
cc1plus: all warnings being treated as errors
make[2]: *** [lib/Target/AMDGPU/Utils/CMakeFiles/LLVMAMDGPUUtils.dir/build.make:90: lib/Target/AMDGPU/Utils/CMakeFiles/LLVMAMDGPUUtils.dir/AMDGPUBaseInfo.cpp.o] Error 1
```

**Comment by svs-quic** [comment]
The function ` llvm::AMDGPU::getMIMGDimInfoByEncoding(uint8_t)`  takes in a unsigned value and hence the check for `(Encoding < 0x0)` will always be false. You are getting this error since you have `-DLLVM_ENABLE_WERROR=ON`  in your cmake command. You can maybe remove this flag from your command to by-pass this error for now.

**Comment by mahesh-attarde** [comment]
Workaround, removing temporarily `-DLLVM_ENABLE_WERROR=ON`  works. 
Thanks @svs-quic
